integrations/airflow/marquez_airflow/extractors/bigquery_extractor.py

```diff
@@ -63,40 +134,52 @@ def _bq_table_name(self, bq_table):
         table = bq_table.get('tableId')
         return f"{project}.{dataset}.{table}"
 
-    def extract_on_complete(self, task_instance) -> StepMetadata:
+    def extract_on_complete(self, task_instance) -> Optional[StepMetadata]:
         log.debug(f"extract_on_complete({task_instance})")
         context = self.parse_sql_context()
-        source = self._source()
 
         try:
             bigquery_job_id = self._get_xcom_bigquery_job_id(task_instance)
-            context['bigquery.job_id'] = bigquery_job_id
             if bigquery_job_id is None:
                 raise Exception("Xcom could not resolve BigQuery job id." +
                                 "Job may have failed.")
         except Exception as e:
             log.error(f"Cannot retrieve job details from BigQuery.Client. {e}",
                       exc_info=True)
-            context['bigquery.extractor.client_error'] = \
-                f"{e}: {traceback.format_exc()}"
             return StepMetadata(
                 name=get_job_name(task=self.operator),
-                context=context,
                 inputs=None,
-                outputs=None
+                outputs=None,
+                run_facets={
+                    "bigQueryError": BigQueryErrorRunFacet(
+                        clientError=f"{e}: {traceback.format_exc()}",
+                        parserError=context.parser_error
+                    )
+                }
             )
 
         inputs = None
-        outputs = None
+        output = None
+        run_facets = {}
         try:
             client = bigquery.Client()
             try:
                 job = client.get_job(job_id=bigquery_job_id)
-                job_properties_str = json.dumps(job._properties)
-                context['bigquery.job_properties'] = job_properties_str
+                props = job._properties
+
+                run_stat_facet, dataset_stat_facet = self._get_output_statistics(props)
+
+                run_facets.update({
+                    "bigQueryStatistics": run_stat_facet
+                })
+
+                inputs = self._get_input_from_bq(props, client)
+                output = self._get_output_from_bq(props, client)
+                if output:
+                    output.custom_facets.update({
+                        "datasetStatistics": dataset_stat_facet
```

Comment:
We should also open a PR in OpenLineage to define this name .
This is not consistent with the Spark integration (`stats`):
https://github.com/MarquezProject/marquez/blob/9e91011ba8072cc72c456df90ec2a5e21e9cb69e/integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/InsertIntoDataSourceVisitor.java#L43
https://github.com/MarquezProject/marquez/blob/9e91011ba8072cc72c456df90ec2a5e21e9cb69e/integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/PlanUtils.java#L150
Which one do we want to keep?
CC @collado-mike

Reply:
This one is a good candidate for standarization.

Reply:
@collado-mike will open a PR on OpenLineage to add it